pkg/store/bucket.go

```diff
@@ -42,6 +42,14 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+// Approximately this is the max number of samples that we may have in any given chunk. This is needed
+// for precalculating the number of samples that we may have to retrieve and decode for any given query
+// without downloading them. Please take a look at https://github.com/prometheus/tsdb/pull/397 to know
+// where this number comes from. Long story short: TSDB is made in such a way, and it is made in such a way
+// because you barely get any improvements in compression when the number of samples is beyond this.
+// Take a look at Figure 6 in this whitepaper http://www.vldb.org/pvldb/vol8/p1816-teller.pdf.
+const maxSamplesPerChunk uint64 = 120
```

Comment:
do we need to allocate uint64? smaller thing would be enough.

Reply:
Will remove the type here so that the compiler can decide itself just like with maxChunkSize.